client/lib/domains/reducer.js

```diff
@@ -82,6 +83,16 @@ function reducer( state, payload ) {
 			} );
 
 		case UpgradesActionTypes.PRIMARY_DOMAIN_SET_COMPLETED:
+			return updateSiteState( state, siteId, {
```

Comment:
It looks a bit complicated but it's just reducer's job. I don't think storing the previous name is a good approach. If we can deduce a data from the current values without adding any extra data, then we should do that. Passing in the previous domain name while you know the previous domain name feels a bit wrong to me.

Besides, even we have the previous domain name, we should see iterate the array to find the match for the domain in the array, and we still have to sort the array so the primary domain is on top.